Update whats-new.md (#381)
harness/developer-hub: release-notes/whats-new.md

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

## January 17, 2023, version 78214

### Platform

> - This release introduces a dedicated release notes page for Harness Delegate &mdash; SaaS. You can find release notes for the NextGen delegate at [Delegate](/release-notes/delegate).

Reviewer: ```suggestion
- This release introduces a dedicated release notes page for Harness delegate &mdash; SaaS. To find release notes for the NextGen delegate, go to the [delegate release notes](/release-notes/delegate).
```

Reviewer: it's not clear if Harness delegate &mdash; SaaS and NextGen delegate are the same thing here, imo. Your call.

Author: I'm going to smooth out the NextGen/SaaS language. But otherwise, I think my references conform with MSTP. I tried to break up information that was in a note on the release notes page so it wasn't a noun stack. As for your suggestions, I like to be clear about the title of the page that I send people to and was taught that people should know where they're going before they jump so I do put that first. MSTP says that referencing the title is fine so I would like to stick with the clarity of using the title. See https://learn.microsoft.com/en-us/style-guide/urls-web-addresses. Also, that "go to" I think is a misinterpretation of a MSTP style guideline that tells people to use "go to" for links to websites instead of "click here". I don't believe it's meant to apply to documentation; instead, I believe it applies to navigation. See https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/n/navigate?source=recommendations.

Reviewer: `go to` is in our style guide
